nomad/client_agent_endpoint.go

```diff
@@ -236,3 +253,59 @@ func (m *Agent) forwardMonitor(conn io.ReadWriteCloser, args cstructs.MonitorReq
 	structs.Bridge(conn, clientConn)
 	return
 }
+
+func (m *Agent) forwardMonitorServer(conn io.ReadWriteCloser, args cstructs.MonitorRequest, encoder *codec.Encoder, decoder *codec.Decoder) {
+	var target *serverParts
+	serverID := args.ServerID
+
+	// empty ServerID to prevent forwarding loop
+	args.ServerID = ""
+
+	if serverID == "leader" {
+		isLeader, remoteServer := m.srv.getLeader()
+		if !isLeader && remoteServer != nil {
+			target = remoteServer
+		}
+		if !isLeader && remoteServer == nil {
+			handleStreamResultError(structs.ErrNoLeader, helper.Int64ToPtr(400), encoder)
+			return
+		}
+	} else {
+		// See if the server ID is a known member
+		serfMembers := m.srv.Members()
+		for _, mem := range serfMembers {
+			if mem.Name == serverID {
+				if ok, srv := isNomadServer(mem); ok {
+					target = srv
+				}
+			}
+		}
+	}
+
+	// Unable to find a server
+	if target == nil {
+		err := fmt.Errorf("unknown nomad server %s", serverID)
+		handleStreamResultError(err, helper.Int64ToPtr(400), encoder)
+		return
+	}
+
+	var serverConn net.Conn
+	localConn, err := m.srv.streamingRpc(target, "Agent.Monitor")
+	if err != nil {
+		handleStreamResultError(err, helper.Int64ToPtr(500), encoder)
+		return
+	}
+
+	serverConn = localConn
```

Comment:
i'm missing something, i think. what's happening with two variables here, `localConn` and `serverConn`?

Reply:
Ah woops, think that was just an oversight/leftover from refactoring

Reply:
aww, i thought i was about to learn some funky new golang hack 😄

Reply:
:D you mean you don't needlessly recreate variables just for fun?